packages/astro/src/content/runtime.ts

```diff
@@ -26,6 +26,13 @@ type CollectionToEntryMap = Record<string, GlobResult>;
 type GetEntryImport = (collection: string, lookupId: string) => Promise<LazyImport>;
 
 export function defineCollection(config: any) {
+	if (
+		('loader' in config && config.type !== CONTENT_LAYER_TYPE) ||
+		(config.type === CONTENT_LAYER_TYPE && !('loader' in config))
+	) {
+		// TODO: when this moves out of experimental, we will set the type automatically
+		throw new AstroError(AstroErrorData.ContentLayerTypeError);
```

Comment:
If you already know if this error is going to be removed when we go out of the experimental, I suggest throwing a normal `Error`. Unfortunately, Astro errors can't be deleted once they are public (that's how the codegen of docs work). I made this very mistake with the rewrite feature

Reply:
Changed to an `AstroUserError` so that when it's caught it propagates our custom message and hint.